src/registers/model_specific.rs

```diff
@@ -172,4 +203,125 @@ mod x86_64 {
             unsafe { Self::MSR.write(address.as_u64()) };
         }
     }
+
+    impl Star {
+        /// Read the Ring 0 and Ring 3 segment bases.
+        /// The remaining fields are ignored because they are
+        /// not valid for long mode.
+        ///
+        /// # Returns
+        /// - Field 1 (SYSRET): The CS selector is set to this field + 16. SS.Sel is set to
+        /// this field + 8. Because SYSRET always returns to CPL 3, the
+        /// RPL bits 1:0 should be initialized to 11b.
+        /// - Field 2 (SYSCALL): This field is copied directly into CS.Sel. SS.Sel is set to
+        ///  this field + 8. Because SYSCALL always switches to CPL 0, the RPL bits
+        /// 33:32 should be initialized to 00b.
+        pub fn read_raw() -> (u64, u64) {
+            let msr_value = unsafe { Self::MSR.read() };
+            let sysret = msr_value.get_bits(48..63);
+            let syscall = msr_value.get_bits(32..47);
+            (sysret, syscall)
+        }
+
+        /// Read the Ring 0 and Ring 3 segment bases.
+        /// Returns
+        /// - CS Selector SYSRET
+        /// - SS Selector SYSRET
+        /// - CS Selector SYSCALL
+        /// - SS Selector SYSCALL
+        pub fn read() -> (
+            SegmentSelector,
+            SegmentSelector,
+            SegmentSelector,
+            SegmentSelector,
+        ) {
+            let raw = Self::read_raw();
+            return (
+                SegmentSelector((raw.0 + 16 - 3).try_into().unwrap()),
+                SegmentSelector((raw.0 + 8 - 3).try_into().unwrap()),
+                SegmentSelector((raw.1).try_into().unwrap()),
+                SegmentSelector((raw.1 + 8).try_into().unwrap()),
+            );
+        }
+
+        /// Write the Ring 0 and Ring 3 segment bases.
+        /// The remaining fields are ignored because they are
+        /// not valid for long mode.
+        ///
+        /// # Parameters
+        /// - sysret: The CS selector is set to this field + 16. SS.Sel is set to
+        /// this field + 8. Because SYSRET always returns to CPL 3, the
+        /// RPL bits 1:0 should be initialized to 11b.
+        /// - syscall: This field is copied directly into CS.Sel. SS.Sel is set to
+        ///  this field + 8. Because SYSCALL always switches to CPL 0, the RPL bits
+        /// 33:32 should be initialized to 00b.
+        pub fn write_raw(sysret: u64, syscall: u64) {
+            let mut msr_value = 0u64;
+            msr_value.set_bits(48..63, sysret);
+            msr_value.set_bits(32..47, syscall);
```

Comment:
This also should be either inclusive ranges (`..=` ) or the upper bound needs to be increased by one (since it's exclusive).